### PR TITLE
Graph theory - minor improvements of Huneke's proof (5)

### PR DESCRIPTION
**Pull request by avekens**
Main results:
* proof of ~numclwlk3lem3, ~extwwlkfablem2 shortened
* proof of ~numclwwlkovf2ex restructured extracting lemmata ~numclwwlkovf2exlem1 and ~numclwwlkovf2exlem2

Auxiliary results:
* ~ccatw2s1cl, ~fzosplitpr moved from AV's mathbox
* ~elfz1uz, ~clwwlksnwwlksn added
* ~clwwlkinwwlk revised
* ~ccatws1lenrev obsolete
* ~extwwlkfablem2lem removed
* proof of ~elfz1b, ~fzosplitprm1 shortened

### CODE REVIEW

File: set.mm
Comment (benjub) [inline set.mm:425100]: as a word (?)

Also: is this true for N e. NN0 ? (I do not know if you consider trivial walks as closed walks)

File: set.mm
Comment (benjub) [inline set.mm:425153]: There is no need for W to be closed for that. (?)

Alos: if I were to write it, I would use N+1 and N for N e. NN0, instead of N and N-1 for N e. NN.  But if later theorems use the other convention, it will take additional steps to "rebase" things.

File: set.mm
Comment (benjub) [inline set.mm:425205]: For Lemmas 1 and 2: can you briefly comment what they prove ? (no need to be too precise since these are only lemmas, so a very short and rough sketch suffices)

File: set.mm
Comment (benjub) [inline set.mm:425355]: If I understand correctly: P (why not "W" ?) is a loop based at X.  If you walk through P, then go to the neighboring vertex Q, and then back to X, you have another loop.  So why does the statement read P + X + Q and not P + Q + X ?
(is a walk represented as a list of vertices?)

File: set.mm
Comment (benjub) [inline set.mm]: Thanks !

File: set.mm
Comment (avekens) [reply set.mm:425100]: OK, I will add the "a".
For N e. NN0,  ( N ClWWalksN G ) is not defined (would be = (/) ), therefore the theorem would not hold.

File: set.mm
Comment (avekens) [reply set.mm:425153]: Here, W is a fixed word, say "abc", and this word is used as representation of the closed walk "abca" of length 3 (3 edges) or as a representation of the (not closed) walk "abc" of length 2 (2 edges). So starting with the closed walk, the length of the corresponding walk is N - 1.

File: set.mm
Comment (avekens) [reply set.mm:425205]: OK, I will add short comments.

File: set.mm
Comment (avekens) [reply set.mm:425355]: Yes, W (for walk) would be better than P (I think for path, which is not true here). I'll see if I can change that without much effort.
Yes, ( X F N) is the set of closed walks of a fixed length N based on a fixed vertex X.
Example: P = "Xabc", representing the closed walk "XabcX", and "QX" is an edge (Q is a neighbor of X), then "XabcXQ" (= P + X + Q) represents the closed walk "XabcXQX"

Comment (benjub) [reply set.mm:425355]: That's strange. Then, how do you know that Xabc is a closed walk, and not a path from X to c ?

Comment (avekens) [reply set.mm:425355]: It depends on the context: <" X a b c " > e. ( 4 ClWWalksN G ) represents a closed walk "XabcX" of length 4, whereas <" X a b c " > e. ( 3 WWalksN G ) represents a (not closed, if X =/= c) walk "Xabc" of length 3. The representation of closed walks by words without repeating the first and last vertex has several advantages, especially for applying the "necklace" theorems about cyclical shifts of words.

File: set.mm
Comment (avekens) [reply set.mm:425205]: done

File: set.mm
Comment (avekens) [reply set.mm:425100]: done

File: set.mm
Comment (avekens) [reply set.mm:425355]: P replaced by W - done.

Comment (benjub) [reply set.mm:425355]: Thanks for the explanation.
And then maybe Q could be Y ?

File: set.mm
Comment (benjub) [inline set.mm]: Q ---> Y (?)
(or maybe X ---> P , depending on your conventions for naming vertices)

**Review by benjub — APPROVED** [review]

File: set.mm
Comment (avekens) [reply set.mm]: I note this for my next PR...

**Comment by jkingdon** [comment]
@avekens I assume you noticed that git conflicts need to be resolved? Hopefully you can get those resolved and we can merge this before more conflicts creep in. (I don't know if you are familiar with resolving git conflicts but let us know if there is anything we can help with).

**Comment by avekens** [comment]
@jkingdon : Hm , I did not see any merge conflicts in Github: neither on creating the PR, nor after other pull requests were merged. I pulled the latest changes and pushed them again now (everything is green in Github - again). I hope that my changes can be merged now without problems.

**Comment by avekens** [comment]
Maybe this PR cannot be merged until @benjub approved it? A (final) review by him was requested.

**Comment by jkingdon** [comment]
> @jkingdon : Hm , I did not see any merge conflicts in Github: neither on creating the PR, nor after other pull requests were merged. I pulled the latest changes and pushed them again now (everything is green in Github - again). I hope that my changes can be merged now without problems.

Oh this was my fault. Github was saying "This branch cannot be rebased due to conflicts" but that was because I had the pulldown set to "Rebase and merge" (which I sometimes do, but only if that particular pull request is broken down into git commits in a way where that makes sense). If I change it to "Squash and merge" then the message goes away.

This normal git/github behavior. The end result of a bunch of commits (if combined into one) might have no conflicts, even if attempting to apply them one by one might have conflicts at some of the intermediate stages.

**Comment by jkingdon** [comment]
> Maybe this PR cannot be merged until @benjub approved it? A (final) review by him was requested.

We don't currently enforce reviews via github (and even if we did I'm not completely sure how that setting would deal with an older approval followed by a request for another review).

As for what to do if the final review comes up with other changes that @benjub and you (@avekens ) want to make, you can always make a small follow-up pull request or include those in a later pull request. Generally speaking that would be my recommendation rather than delaying the whole pull request.

**Comment by david-a-wheeler** [comment]
> We don't currently enforce reviews via github (and even if we did I'm not completely sure how that setting would deal with an older approval followed by a request for another review).

Details are here:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews

Basically, there's a separate setting called "stale review dismissal". If enabled, then previous approval is dismissed & the revised change has to be approved again to count.

**Pull request merged**
